catkit/hardware/flir/flir_camera.py

```diff
@@ -0,0 +1,356 @@
+    pixel_format = _create_enum_property('PixelFormat', 'pixel_format_enum')
+
+    # Bit depth of the ADC (8bit, 12bit, etc...)
+    adc_bit_depth = _create_enum_property('AdcBitDepth', 'adc_bit_depth_enum')
+
+    # Allows for modifying the duty cycle of the camera. If disabled, the camera runs
+    # at 100% duty cycle (excluding sensor readout), regardless of the value of
+    # `acquisition_frame_rate`.
+    acquisition_frame_rate = _create_property('AcquisitionFrameRate')
+    acquisition_frame_rate_enable = _create_property('AcquisitionFrameRateEnable')
+
+    @property
+    def device_name(self):
+        '''The model of the camera (eg. "BFS-U3-04S2M-C").
+        '''
+        return self.cam.TLDevice.DeviceModelName.GetValue()
+
+    def stream_exposures(self, exposure_time, num_exposures, extra_metadata=None):
+        '''Stream exposures from the camera.
+
+        Note: contrary to the ZWO camera, all subarray settings must be set on the
+        camera before this generator is called.
+
+        Parameters
+        ----------
+        exposure time : scalar or pint quantity
+            The exposure time for each frame (in us if this is a scalar).
+        num_exposures : int
+            The number of exposures to yield. It is safe to stop this generator before
+            this number of exposures is reached.
+        extra_metadata : list of MetaDataEntry objects
+            The metadata to attach to each output frame.
+
+        Yields
+        ------
+        img : ndarray
+            Each camera frame separately.
+        meta_data : list of MetaDataEntry objects
+            The metadata for each frame.
+        '''
+        if not type(exposure_time) in [int, float]:
+            exposure_time_us = exposure_time.to(units.microsecond).m
+        else:
+            exposure_time_us = exposure_time
+
+        self.exposure_time = exposure_time_us
+
+        meta_data = [MetaDataEntry("Exposure Time", "EXP_TIME", exposure_time_us, "microseconds")]
+        meta_data.append(MetaDataEntry("Camera", "CAMERA", self.config_id, "Camera name, correlates to entry in ini"))
+        meta_data.append(MetaDataEntry("CameraModel", "MODEL", self.device_name, "Camera model name"))
+        meta_data.append(MetaDataEntry("Gain", "GAIN", self.gain, "Gain for camera"))
+
+        if extra_metadata is not None:
+            if isinstance(extra_metadata, list):
+                meta_data.extend(extra_metadata)
+            else:
+                meta_data.append(extra_metadata)
+
+        # try-finally shuts down the camera if the generator is stopped.
+        try:
+            self.cam.BeginAcquisition()
+            frame_count = 0
+
+            # Get the pixel format to interpret the raw data.
+            if self.pixel_format == 'mono8':
+                pixel_format = self.instrument_lib.PixelFormat_Mono8
+            else:
+                pixel_format = self.instrument_lib.PixelFormat_Mono16
```

Comment:
[revision] I could be wrong but this entire thing seems very circular and cries for an enum, you've implemented a similar mapping more than once now.

I would make an enum and then just assign a member to ``self.pixel_format`` and be done with it.